### ISSUE TITLE
Should we keep the Inject/Extract extension methods for formats?

### ISSUE THREAD

**Issue by cwe1ss** [opentracing/opentracing-csharp#25]
Analogous to known tags ( #18 ), we currently also have `ITracer` extension methods for the built-in formats.

Examples:
```csharp
public static void InjectTextMap(this ITracer tracer, ISpanContext spanContext, IDictionary<string, string> data)
{
    tracer.Inject(spanContext, Formats.TextMap, new DictionaryCarrier(data));
}

public static ISpanContext ExtractTextMap(this ITracer tracer, IDictionary<string, string> data)
{
    return tracer.Extract(Formats.TextMap, new DictionaryCarrier(data));
}

// usage

var data = new Dictionary<string, string>();

tracer.InjectTextMap(existingSpanContext, data);

var extractedSpanContext = tracer.ExtractTextMap(data);
```

The main advantages are:
* Users can pass the actual object instead of having to manually wrap it in a carrier
  * This means, they don't need to know/search for the wrapper types
* People who write their own formats/carrier types can also create their own extension methods so they will show up in the same IntelliSense tooltip.

Are you ok with this or do you see any concerns?

/cc @bensigelman @dawallin @yurishkuro

**Comment by yurishkuro** [comment]
I don't have a strong opinion on these. I do have misgivings in general about overloading the lib with helper functions, and from the past experience with C# I remember that extension methods are a bit hard to follow when reading the code as they are often imported inconspicuously.

**Comment by bensigelman** [comment]
I can see advantages here, but my personal opinion is that any convenience functions should be kept out of the OT layer until there's a clear pattern/problem emerging in actual production calling code. My two cents.

**Comment by cwe1ss** [comment]
Extension methods have been removed with #26 - thank you for your feedback!
